src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import static duke.Event.DATE_TIME_FORMATTER;
+import static duke.Storage.saveTasksToFile;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import javafx.application.Platform;
+
+
+
+
+/**
+ * Handles the parsing of user input and the corresponding actions in the Duke application.
+ */
+public class Parser {
+    static final String EXIT_PHRASE = "bye";
+    static final String LIST_PHRASE = "list";
+    static final String TODO_PHRASE = "todo";
+    static final String DEADLINE_PHRASE = "deadline";
+    static final String EVENT_PHRASE = "event";
+    static final String DELETE_PHRASE = "delete";
+    static final String SEARCH_PHRASE = "find";
+
+
+    /**
+     * Parses the user input and performs the corresponding actions.
+     *
+     * @param taskList The list of tasks.
+     * @param i The current index of tasks in the list.
+     * @param ui The user interface object.
+     * @throws DukeException If an error occurs during parsing.
+     */
+    public static String parse(ArrayList<Task> taskList,
+                             int i, Ui ui, Storage storage, String userInput) throws DukeException {
+        String finalText = "";
+        String[] binaryArr = userInput.split(" ", 2);
+        if (binaryArr.length == 1) {
+            binaryArr = new String[] {binaryArr[0], null};
+        }
+
+        String commandWord = binaryArr[0];
+        if (commandWord.equals(EXIT_PHRASE)) {
+            saveTasksToFile(taskList, String.valueOf(storage.path));
+            Platform.exit();
+            return "Bye. Hope to see you again soon!";
+        } else if (commandWord.equals(LIST_PHRASE)) {
+            if (taskList.isEmpty()) {
+                finalText = "There are currently no tasks in your list";
+            } else {
+                finalText += "Here are the tasks in your list:\n";
+                for (int j = 0; j < i; j++) {
+                    finalText += j + 1 + "." + taskList.get(j).toString() + "\n";
+                }
+            }
+            return finalText;
+        } else if (commandWord.equals("mark")) {
```

Comment:
Should these be defined as constants consistency since the other commands are also stored in constants?